proxy/src/main.rs

```diff
@@ -166,6 +168,10 @@ async fn main() -> anyhow::Result<()> {
     // Furthermore, the first one to fail will cancel the rest.
     let _: Vec<()> = futures::future::try_join_all(tasks).await?;
 
+    // Shutdown trace pipeline gracefully, so that it has a chance to send any
+    // pending traces before we exit.
+    tracing_utils::shutdown_tracing();
```

Comment:
There's no graceful shutdown in the proxy, so this is in vain. You could leave a TODO here instead.